### DISCUSSION TITLE
Integration of web searches

### DISCUSSION THREAD

**Discussion by rthwm** [LostRuins/koboldcpp#1268] [Ideas]
Love Koboldcpp and how it works! I was needing a means to have the AI access live/current data from the web. After installing text-generation-webui with the extension "LLM_Web_search", I was able to get it to work almost perfectly. I just really don't like the UI and much prefer KoboldCPP. Just curious if there might be a way to integrate this feature into a future release.

**Comment by LostRuins** [comment]
Now Added in 1.81

**Reply by VL4DST3R** [reply]
How do you actually trigger it to querry the web? I tried the example you provided in the release page and it did indeed work, but I can't seem to get it to look up any other info (opting to either hallucinate or use present/outdated info)

**Reply by LostRuins** [reply]
You need to make sure it's enabled in Kcpp, and then toggled on in the correct menu.

![image](https://github.com/user-attachments/assets/bb971941-624b-4d61-81fb-4edae681d8fd)

It will make a query to DuckDuckGo with your instruction, scrape the top 3 results and include them in the context.

**Reply by VL4DST3R** [reply]
I mentioned that it did work (i.e. I have properly enabled it) when using your example, but I'm not sure how/what to prompt to make it do it again, for other topics. Or if it is supposed to work every time/prompt after enabling it, it did not work for me on any subsequent messages.

**Reply by LostRuins** [reply]
Please run it in debugmode and take a look at what is displayed in the terminal.

**Reply by LostRuins** [reply]
![image](https://github.com/user-attachments/assets/1db2c114-a4f9-457e-86b2-cc631c59a808)

![image](https://github.com/user-attachments/assets/c6bd576d-3ce7-4dc6-8173-494f343fdcbe)

You can see the search query triggered, and the result embedded into the prompt

**Reply by rthwm** [reply]
First off, thank you for the integration! I am experimenting with it and it does appear to be working (I can see the web sources in the logs that it is pulling information from). Will need to experiment with the output's a little bit. When I tried to get it to tell me the current date and/or current time, I can see the sources that it is pulling from, but the information that is given in the response doesn't match what the sources are actually telling it. ex: "what is today's date in central united states". Logs shows it did a search and found answers on 24timezones.com and timeanddate.com, which both sites do give the correct date/time information, but the LLM is returning the wrong date/time from what the sources are telling it.

**Reply by VL4DST3R** [reply]
I am curious regarding the websearch: how does it manage to detect when it should perform a search and when not?

**Reply by LostRuins** [reply]
You have to toggle it on/off. There's now a button to do so.

**Reply by VL4DST3R** [reply]
Got it. In the old initial release it seemed to search only sometimes, so I left thinking there was more to it but now indeed it seems to be consistently searching (or not) depending on the button state, which is very straightforward, thank you.
